website/docs/docs/collaborate/govern/model-versions.md

```diff
@@ -9,32 +9,76 @@ description: "Version models to help with lifecycle management"
+
+## When should you version a model?
+
+Many changes to a model are not breaking, and do not require a new version! Examples include adding a new column, or fixing a bug in modeling logic.
+
+By enforcing a model's contract, dbt can help you catch unintended changes to column names and data types that could cause a big headache for downstream queriers.
+
+It's also possible to change the model in more subtle ways — by recalculating a column in a way that doesn't change its name, data type, or enforceable characteristics—but would substantially change the results seen by downstream queriers.
+
+The process of sunsetting and migrating model versions requires real work, and may require significant coordination across teams. If, instead of using model versions, you opt for non-breaking changes wherever possible—that's a completely legitimate approach. Even so, after a while, you'll find yourself with lots of unused or deprecated columns. Many teams will want to consider a predictable cadence (once or twice a year) for bumping the version of their mature models, and taking the opportunity to remove no-longer-used columns.
 
 ## How is this different from "version control"?
 
 [Version control](git-version-control) allows your team to collaborate simultaneously on a single code repository, manage conflicts between changes, and review changes before deploying into production. In that sense, version control is an essential tool for versioning the deployment of an entire dbt project—always the latest state of the `main` branch, with the ability to "rollback" changes by reverting a commit or pull request. In general, only one version of your project code is deployed into an environment at a time.
 
 Model versions are different. Multiple versions of a model will live in the same code repository at the same time and be deployed into the same data environment simultaneously. This is similar to how web APIs are versioned—multiple versions are live simultaneously; older versions are often eventually sunsetted.
 
-dbt's model `versions` makes it possible to define multiple versions:
-- That share the same "reference" name
-- While reusing the same top-level properties, highlighting just their differences
+## How is this different from just creating a new model?
+
+Honestly, it's only a little bit different! There isn't much magic here, and that's by design.
+
+You've always been able to create a new model, and name it `dim_customers_v2`. Why should you opt for a "real" versioned model instead?
+
+First, the versioned model preserves its _reference name_. Versioned models are `ref`'d by their _model name_, rather than the name of the file that they're defined in. By default, the `ref` resolves to the latest version (as declared by that model's maintainer), but you can also `ref` a specific version of the model, with a `version` keyword.
```

Comment:
Does `latest` version of a model get to look for the `_vX`-less sql file? I don't like that the grid here forces new sql files for every version, so [you don't get a nice git diff](https://github.com/dbt-labs/dbt-core/discussions/6736#:~:text=This%20also%20simplifies%20the%20git%20and%20code%20review%20flow%20%2D%20when%20you%20bump%20a%20version%2C%20you%20are%20comparing%20the%20old%20version%20of%20the%20file%20to%20the%20new%20version%20of%20the%20same%20file%20instead%20of%20doing%20a%20diff%20between%20dim_customers_v1.sql%20and%20dim_customers_v2.sql)

Reply:
> Does `latest` version of a model get to look for the _vX-less sql file?

That's not the case in the current implementation. Should it be?

You could do the same thing as with aliases, and keep moving the `defined_in` property around, so that `dim_customers.sql` is always your "latest":

```yml
models: 
  - name: dim_customers
    latest_version: 2
    versions:
      - v: 3
        defined_in: dim_customers_NOT_READY_YET.sql
        ...
      - v: 2
        # because this is the latest, it should have the canonical file name + alias
        alias: dim_customers
        defined_in: dim_customers
        ...
      - v: 1
        ...      
```

Reply:
It feels weird/WET to _have_ to move both the `alias` and `defined_in` around over time to just get the same name as is already defined in the `name` key up top